app/models/mixins/relationship_mixin.rb

```diff
@@ -352,45 +349,44 @@ def descendants_arranged(*args)
   # Returns a list of all relationships in the record's subtree
   def subtree_rels(*args)
     options = args.extract_options!
+    # TODO: make this a single query (vs 3)
+    # thus making filter_by_resource_type into a query
     rels = relationships.flat_map(&:subtree).uniq
+    # empty tree returns self
```

Comment:
Please remove this comment as it sort of should belong everywhere we have this `Relationship.new(:resource => self)`, but is only here.